src/libs/Visibility/index.desktop.js

```diff
@@ -0,0 +1,38 @@
+import ELECTRON_EVENTS from '../../../desktop/ELECTRON_EVENTS';
+
+/**
+ * Detects whether the app is visible or not. Electron supports
+ * document.visibilityState, but switching to another app while
+ * Electron is partially occluded will not trigger a state of hidden
+ * so we ask the main process synchronously whether the
+ * BrowserWindow.isFocused()
+ *
+ * @returns {Boolean}
+ */
+function isVisible() {
+    return window.electron.sendSync(ELECTRON_EVENTS.REQUEST_VISIBILITY);
+}
+
+/**
+ * Adds event listener for changes in visibility state
+ *
+ * @param {Function} callback
+ *
+ * @return {Object} To have .remove() invoked to remove listener
+ */
+function addEventListener(callback) {
+    window.electron.on(ELECTRON_EVENTS.FOCUS, callback);
+    window.electron.on(ELECTRON_EVENTS.BLUR, callback);
```

Comment:
I think it might be possible for someone to assume that since `AppState` calls the callback with a `state` argument on native that it would do the same for desktop - and probably this is sending some other kind of event (though didn't check). 

For consistency, it would be good if all of these methods just called the callback with either:

- No arguments at all if the intended usage is to trigger the callback for any change generically
- Call the callback with `isVisible` (not sure if we need to do this)